src/lib.rs

```diff
@@ -373,6 +395,83 @@ pub fn return_fdtable_copy(cageid: u64) -> HashMap<u64, FDTableEntry> {
     fdtable.get(&cageid).unwrap().clone()
 }
 
+/* SELECT()
+*   fd_set is used in the Linux select system call to specify the file descriptor 
+*   to be monitored. fd_set is actually a bit array, each bit of which represents 
+*   a file descriptor. We use Vec to express the fd_set of the virtual file descriptor 
+*   in Lind, and expand the conversion function between lind fd_set and kernel fd_set.
+*
+*   We won't change value inside of Vec, so only using borrow immutable reference
+*   for virtual_to_real_set
+*   Reference: https://github.com/rust-lang/libc/issues/475
+*/
+pub fn virtual_to_real_set(cageid: u64, virtualfds: &Vec<u64>) -> fd_set {
+    let fdtable = GLOBALFDTABLE.lock().unwrap();
+    if !fdtable.contains_key(&cageid) {
+        panic!("Unknown cageid in fdtable access");
+    }
+
+    let mut real_set = unsafe { std::mem::zeroed::<fd_set>() };
+
+    for &virtualfd in virtualfds {
+        let real_fd = translate_virtual_fd(cageid, virtualfd).unwrap() as RawFd;
+        unsafe { FD_SET(real_fd, &mut real_set) };
```

Comment:
What is this doing and why?

Reply:
`fd_set` is a bit array. We get the kernel file descriptor according to the virtual file descriptor we received, and use `FD_SET` to set that into the `fd_set`.

Reply:
From my understanding, `fd_set` is a specific data type used by the kernel, so we need to make sure the final variable we pass to the kernel is in the format that the kernel expects. That's why I choose to use `FD_SET` function instead of doing bitmask by myself

Reply:
Is it better to do this conversion later, after we've released the lock?  Do we need the lock for this?

Reply:
Yeah preview code will make program hang, and I fixed that in the newest commit